scrapy/extensions/spiderstate.py

```diff
@@ -12,7 +13,11 @@ def __init__(self, jobdir=None):
 
     @classmethod
     def from_crawler(cls, crawler):
-        obj = cls(job_dir(crawler.settings))
+        jobdir = job_dir(crawler.settings)
+        if not jobdir:
```

Comment:
Also [here](https://github.com/scrapy/scrapy/blob/master/scrapy/utils/job.py#L5) it prevents it.